tests/v2/validation/provisioning/rke2/templates_test.go

```diff
@@ -0,0 +1,164 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	localCluster   = "local"
+	fleetNamespace = "fleet-default"
+)
+
+type ClusterTemplateTest struct {
+	Repo             *v1.ClusterRepo `json:"repo,omitempty" yaml:"repo,omitempty"`
+	TemplateName     string          `json:"templateName,omitempty" yaml:"templateName,omitempty"`
+	TemplateProvider string          `json:"templateProvider,omitempty" yaml:"templateProvider,omitempty"`
+}
+
+type ClusterTemplateTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	templateConfig     *ClusterTemplateTest
+	cloudCredentials   *cloudcredentials.CloudCredential
+}
+
+func (r *ClusterTemplateTestSuite) TearDownSuite() {
+	r.session.Cleanup()
+}
+
+func (r *ClusterTemplateTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+	r.templateConfig = new(ClusterTemplateTest)
+	config.LoadConfig("templateTest", r.templateConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+	r.client = client
+
+	enabled := true
+	var testuser = namegenerator.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	provider := provisioning.CreateProvider(r.templateConfig.TemplateProvider)
+	r.cloudCredentials, err = provider.CloudCredFunc(client)
+	require.NoError(r.T(), err)
+}
+
+func (r *ClusterTemplateTestSuite) TestProvisionRKE2TemplateCluster() {
+	_, err := r.client.Catalog.ClusterRepos().Create(context.TODO(), r.templateConfig.Repo, metav1.CreateOptions{})
+	require.NoError(r.T(), err)
+
+	watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + r.templateConfig.Repo.Name,
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	require.NoError(r.T(), err)
+
+	err = wait.WatchWait(watchRepo, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Added {
+			return true, nil
+		} else if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error waiting for repo to become active")
+		}
+		return false, nil
+	})
+	require.NoError(r.T(), err)
+
+	r.client.Session.RegisterCleanupFunc(func() error {
+		err := r.client.Catalog.ClusterRepos().Delete(context.TODO(), r.templateConfig.Repo.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return err
+		}
+
+		watchRepo, err := r.client.Catalog.ClusterRepos().Watch(context.TODO(), metav1.ListOptions{
```

Comment:
Can we move this to a function - similar to one I suggested above ^, verifyRemoveRepo or something relevant?